### PR TITLE
tests: add tests for `SHA1`

### PR DESCRIPTION
**Pull request by vil02**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

This PR adds test for [`SHA1`](https://github.com/TheAlgorithms/JavaScript/blob/cf0004190ab69c947d781f5cb8537b446beb8cbf/Hashes/SHA1.js). The test data comes from [wikipeda](https://en.wikipedia.org/wiki/SHA-1) or was verified with http://www.sha1-online.com/

- [x] Add missing tests.

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Hashes/tests/SHA1.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    {
+      args: [
+        '12345678901234567890123456789012345678901234567890123456789012345678901234567890'
+      ],
+      expected: '50abf5706a150990a08b2c5ea40fa0e585554732'
+    }
+  ]
+
+  TEST_CASES.forEach((testCase) => {
```
Comment (appgurueu) [inline Hashes/tests/SHA1.test.js:32]: Why not use vitest's `it.each`?

File: Hashes/tests/SHA1.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    it(`check with ${JSON.stringify(testCase.args)}`, () => {
+      const result = SHA1.apply(this, testCase.args)
```
Comment (appgurueu) [inline Hashes/tests/SHA1.test.js:34]: Why not simply `SHA1(testCase.arg)`? Then you could replace `args: [x]` with `arg: x`.

**Review by appgurueu — DISMISSED** [review]
Otherwise fine.

File: Hashes/tests/SHA1.test.js (hunk 1 above)
Comment (vil02) [reply Hashes/tests/SHA1.test.js]: Done in 96c5567f0434e3713365a13b1c6b142a8b39d9fe.

Thanks for this tip. I was not aware of this _syntax_. These are my first lines of JavaScript (not that I want to learn it, I just want to try it).

File: Hashes/tests/SHA1.test.js (hunk 2 above)
Comment (vil02) [reply Hashes/tests/SHA1.test.js]: As above: I did not know about that. In any case, this is outdated by 96c5567f0434e3713365a13b1c6b142a8b39d9fe.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**